From 4c95feb1726a202a7ab7c1f1d16eeb5a0c3dd485 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:04 -0400
Subject: [fs] nfsd: Fix MACH_CRED NULL dereference

Message-id: <1403560314-32525-3-git-send-email-bfields@redhat.com>
Patchwork-id: 84450
O-Subject: [Patch RHEL7 002/172] nfsd4: Fix MACH_CRED NULL dereference
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Fixes a NULL-dereference on attempts to use MACH_CRED protection over
auth_sys.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: c47205914cf5a8cf564ca560aa6eaa2287867e58
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index de393db..4704736 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -1268,6 +1268,8 @@ static bool svc_rqst_integrity_protected(struct svc_rqst *rqstp)
  struct svc_cred *cr = &rqstp->rq_cred;
  u32 service;
 
+ if (!cr->cr_gss_mech)
+  return false;
  service = gss_pseudoflavor_to_service(cr->cr_gss_mech, cr->cr_flavor);
  return service == RPC_GSS_SVC_INTEGRITY ||
         service == RPC_GSS_SVC_PRIVACY;
-- 
1.7.1